qa/c8-orchestration-cluster-e2e-test-suite/pages/TaskPanelPage.ts

```diff
@@ -47,44 +47,43 @@ class TaskPanelPage {
     // Mapping of processes names to process IDs for V2 mode compatibility
     const processNameToIdMapping: Record<string, string> = {
       'User registration': 'user_registration',
-      'Some user activity': 'usertask_to_be_completed', // Process ID from BPMN
-      usertask_to_be_completed: 'usertask_to_be_completed', // Allow process ID as input too
+      'Some user activity': 'usertask_to_be_completed',
       'User registration with vars': 'user_registration_with_vars',
       'User Task with form rerender 1': 'user_task_with_form_rerender_1',
       'User Task with form rerender 2': 'user_task_with_form_rerender_2',
-      UserTask_Number: 'UserTask_Number',
       'Date and Time Task': 'Date_and_Time_Task',
       'Checkbox Task': 'Checkbox_User_Task',
       'Select User Task': 'Select',
       'Radio Button Task': 'Radio_Button_User_Task',
       'Checklist User Task': 'Checklist_Task',
       'Tag list Task': 'Tag_List_Task',
-      Text_Templating_Form_Task: 'Text_Templating_Form_Task',
-      processWithDeployedForm: 'processWithDeployedForm',
-      Zeebe_user_task: 'Zeebe_user_task',
     };
 
-    // Strategy 1: Try with original name (for V1)
-    try {
-      await this.availableTasks
+    const processId = processNameToIdMapping[name] || name;
+
+    const winner = await Promise.race([
```

Comment:
Please create a separate page object for each version as I told you

There's nothing that guarantees that the node won't disappear during a polling between the `await` from the `Promise.race` and the second selection and second selection after you decided the winner

Reply:
I saw your message about the workaround only now.

This should be a better solution I think, please check if it works

```ts
  const taskItem = await Promise.race([
      this.availableTasks
        .getByText(name, {exact: true})
        .nth(0)
        .waitFor({state: 'visible', timeout: 5000}),
      this.availableTasks
        .getByText(processId, {exact: true})
        .nth(0)
        .waitFor({state: 'visible', timeout: 5000})
        
    ]).catch(() => {
      console.log('Failed to open task with name:', name);
      throw new Error(`Task not found: ${name}`);
    });

    await taskItem.click();
```